File: easy/1925_count_square_sum_triples/solution.py
import math


class Solution:
    def countTriples(self, n: int) -> int:
        triples_count = 0

        for a in range(1, n + 1):
            for b in range(1, n + 1):
                c = math.sqrt(a * a + b * b)

                if c.is_integer() and c <= n:
                    triples_count += 1

        return triples_count

File: easy/1925_count_square_sum_triples/solution.cpp
using namespace std;

#include <cmath>

class Solution {
public:
  int countTriples(int n) {
    int triplesCount = 0;
    double c;

    for (int a = 1; a <= n; ++a) {
      for (int b = 1; b <= n; ++b) {
        c = sqrt(a * a + b * b);

        if (c == (int)c && c <= n)
          triplesCount++;
      }
    }

    return triplesCount;
  }
};
